src/main/java/com/google/cloud/genomics/dataflow/utils/BreakFusionTransform.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.genomics.dataflow.utils;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.GroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.Keys;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+
+/*
+ * Breaks DataFlow fusion by doing GroubByKey/Ungroup that forces materialization of the data,
+ * thus preventing DataFlow form fusing steps before and after this transform.
+ * This is useful to insert in cases where a series of transforms deal with very small sets of data
+ * that act as descriptors of very heavy workloads in subsequent steps (e.g. a collection of file names
+ * where each file takes a long time to process).
```

Comment:
That makes sense.  Basically the reason I asked is that subsequently ([in lines 30-31](https://github.com/iliat/dataflow-java/blob/9fa7583736bd2422551a2ab0c4c0d016f22cfeba/src/main/java/com/google/cloud/genomics/dataflow/utils/BreakFusionTransform.java#L30-L31)) it's mentioned that:

 > In this case DataFlow might over-eagerly fuse steps dealing with small datasets with the "heavy" processing steps, which will result in heavy steps being executed on a single worker.

 I interpreted the word "with" as a necessary conjunction.  This indicated to me that the fusion is not only based on the small dataset size only - which would be nice to define as what is small - but rather that the fusion is also based on the heavy step in conjunction, which would indicate a job resubmit on less workers.